### PR TITLE
WIP: Implement Service router, remove ServiceBroker

### PR DESCRIPTION
**Pull request by jtwaleson**
Not sure if the classes now end up in the right files. Any suggestions welcome.

### CODE REVIEW

**Comment by jtwaleson** [comment]
@eruvanos any thoughts?

File: openbrokerapi/api.py [hunk 1]
```diff
@@ -84,6 +86,25 @@ def decorated(*args, **kwargs):
...
 
+    def get_service_by_id(service_id: str):
+        for service in services:
+            if service.id == service_id:
+                return service
+        raise KeyError('service {} not found'.format(service_id))
+
+    def add_service_id_to_async_response(response, service_id: str):
+        patch = False
+        if isinstance(response, ProvisionedServiceSpec):
```
Comment (eruvanos) [inline openbrokerapi/api.py]: We could also add a `@property` `is_async` to ProvisionedServiceSpec, so we avoid here checking for 'isinstance'.

Would also shorter the code.
`patch = response.is_async`

File: openbrokerapi/response.py [hunk 2]
```diff
@@ -16,12 +15,16 @@ def __init__(self,
         self.description = description
 
 
+class AsyncResponse:
+    pass
+
```
Comment (eruvanos) [inline openbrokerapi/response.py]: shows clearly, what could be async, that's nice. I always double checked with the spec ^^

File: openbrokerapi/api.py (hunk 1 above)
Comment (jtwaleson) [reply openbrokerapi/api.py]: It's currently a bit dirty as we call this on separate types of classes, but at least the check (isinstance) is dirty too. I was still looking for a better way to only send Response or Spec classes into this method.

**Comment by jtwaleson** [comment]
Thanks for merging! I was not happy with the last test (catalog) yet, the current solution does not really make sense. I think it should be approved.

**Pull request merged**